File: src/main/webapp/angular/services/roomService.js
'use strict';

application.service('roomService', [
  '$http',
  function ($http) {
    this.findAll = () => $http.get('/iaa-housework/api/room');
    this.saveRoom = (event) => $http.post('/iaa-housework/api/room', event);
  }
]);

File: src/main/java/de/nak/iaa/housework/service/RoomService.java
package de.nak.iaa.housework.service;

import java.time.LocalDateTime;
import java.util.List;

import de.nak.iaa.housework.model.Room;

/**
 * Spezialisierung des allgemeinen {@link DomainService}. Angereichtert um Raum-spezifische Methoden.
 * 
 * @author da0015 14096
 *
 */
public interface RoomService extends DomainService<Room> {

	/**
	 * Liefert die noch nicht belegten Räume in einem Intervall
	 * 
	 * @param start des Intervalls
	 * @param end des Intervalls
	 * @return die verfügbaren Räume
	 */
	List <Room> getAvailableRooms (LocalDateTime start, LocalDateTime end);
}
